### PR TITLE
Changes to Send Success and Error ACK

### PR DESCRIPTION
**Pull request by guruchandru**

### CODE REVIEW

**Review by shilpa24balaji — APPROVED** [review]

File: src/xmidtsend_rbus.c [hunk 1]
```diff
@@ -436,7 +436,7 @@ void sendXmidtEventToServer(wrp_msg_t * msg, rbusMethodAsyncHandle_t asyncHandle
 	{
 		errorMsg = strdup("Memory allocation failed");
 		ParodusError("Memory allocation failed\n");
-		createOutParamsandSendAck(msg, asyncHandle, errorMsg, MSG_PROCESSING_FAILED);
+		createOutParamsandSendAck(msg, asyncHandle, errorMsg, MSG_PROCESSING_FAILED, RBUS_ERROR_INVALID_INPUT);
```
Comment (sadhyama) [inline src/xmidtsend_rbus.c:439]: same as above, here also issue happened due to provider processing failure .
so error code should be RBUS_ERROR_INVALID_RESPONSE_FROM_DESTINATION

File: src/xmidtsend_rbus.c [hunk 2]
```diff
@@ -64,7 +64,7 @@ void addToXmidtUpstreamQ(wrp_msg_t * msg, rbusMethodAsyncHandle_t asyncHandle)
 	{
 		char * errorMsg = strdup("Max Queue Size Exceeded");
 		ParodusError("Queue Size Exceeded\n");
-		createOutParamsandSendAck(msg, asyncHandle, errorMsg , QUEUE_SIZE_EXCEEDED);
+		createOutParamsandSendAck(msg, asyncHandle, errorMsg , QUEUE_SIZE_EXCEEDED, RBUS_ERROR_INVALID_INPUT);
```
Comment (sadhyama) [inline src/xmidtsend_rbus.c:67]: @shilpa24balaji Max queue size is invalid input from t2 ? or it should be an error >100 with "RBUS_ERROR_INVALID_RESPONSE_FROM_DESTINATION" as it happened due to client disconnected for too long. 

https://github.com/xmidt-org/parodus/blob/master/src/xmidtsend_rbus.h#L54

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/xmidtsend_rbus.c [hunk 3]
```diff
@@ -182,7 +182,7 @@ int processData(wrp_msg_t * msg, rbusMethodAsyncHandle_t asyncHandle)
 	{
 		ParodusError("xmidtMsg is NULL\n");
 		errorMsg = strdup("Unable to enqueue");
-		createOutParamsandSendAck(xmidtMsg, asyncHandle, errorMsg, ENQUEUE_FAILURE);
+		createOutParamsandSendAck(xmidtMsg, asyncHandle, errorMsg, ENQUEUE_FAILURE, RBUS_ERROR_INVALID_INPUT);
```
[the remark this answers is not in the document]
Comment (guruchandru) [reply src/xmidtsend_rbus.c]: resolved in latest code

File: src/xmidtsend_rbus.c [hunk 4]
```diff
@@ -371,7 +371,7 @@ void sendXmidtEventToServer(wrp_msg_t * msg, rbusMethodAsyncHandle_t asyncHandle
 		{
 			ParodusError("wrp msg_len is zero\n");
 			errorMsg = strdup("Wrp message encoding failed");
-			createOutParamsandSendAck(msg, asyncHandle, errorMsg, WRP_ENCODE_FAILURE);
+			createOutParamsandSendAck(msg, asyncHandle, errorMsg, WRP_ENCODE_FAILURE, RBUS_ERROR_INVALID_INPUT);
```
[the remark this answers is not in the document]
Comment (guruchandru) [reply src/xmidtsend_rbus.c]: resolved in latest code

File: src/xmidtsend_rbus.c (hunk 1 above)
Comment (guruchandru) [reply src/xmidtsend_rbus.c]: resolved in latest code

File: src/xmidtsend_rbus.c (hunk 2 above)
Comment (shilpa24balaji) [reply src/xmidtsend_rbus.c:67]: Ya lets map max queue size as RBUS_ERROR_INVALID_RESPONSE_FROM_DESTINATION, as it is the response from parodus not being able to process due to wan down etc.

Comment (guruchandru) [reply src/xmidtsend_rbus.c]: Resolved in latest change

**Pull request merged**